drivers/interrupt_controller/exti_stm32.c

```diff
@@ -117,88 +118,24 @@ void stm32_exti_enable(int line)
 		 */
 		return;
 	}
-#elif CONFIG_SOC_SERIES_STM32F1X
-	if (line >= 5 && line <= 9) {
-		irqnum = EXTI9_5_IRQn;
-	} else if (line >= 10 && line <= 15) {
-		irqnum = EXTI15_10_IRQn;
-	} else {
-		/* pins 0..4 are mapped to EXTI0.. EXTI4 */
-		irqnum = EXTI0_IRQn + line;
-	}
-#elif CONFIG_SOC_SERIES_STM32F2X
-	if (line >= 5 && line <= 9) {
-		irqnum = EXTI9_5_IRQn;
-	} else if (line >= 10 && line <= 15) {
-		irqnum = EXTI15_10_IRQn;
-	} else if (line >= 0 && line <= 4) {
-		/* pins 0..4 are mapped to EXTI0.. EXTI4 */
-		irqnum = EXTI0_IRQn + line;
-	} else {
-		switch (line) {
-		case 16:
-			irqnum = PVD_IRQn;
-			break;
-		case 17:
-			irqnum = RTC_Alarm_IRQn;
-			break;
-		case 18:
-			irqnum = OTG_FS_WKUP_IRQn;
-			break;
-		case 21:
-			irqnum = TAMP_STAMP_IRQn;
-			break;
-		case 22:
-			irqnum = RTC_WKUP_IRQn;
-			break;
-		}
-	}
-#elif CONFIG_SOC_SERIES_STM32F3X
+#elif defined(CONFIG_SOC_SERIES_STM32F1X) || \
+      defined(CONFIG_SOC_SERIES_STM32F2X) || \
+      defined(CONFIG_SOC_SERIES_STM32F3X) || \
+      defined(CONFIG_SOC_SERIES_STM32F4X) || \
+      defined(CONFIG_SOC_SERIES_STM32F7X) || \
+      defined(CONFIG_SOC_SERIES_STM32L4X)
 	if (line >= 5 && line <= 9) {
 		irqnum = EXTI9_5_IRQn;
 	} else if (line >= 10 && line <= 15) {
 		irqnum = EXTI15_10_IRQn;
-	} else {
-		/* pins 0..4 are mapped to EXTI0.. EXTI4 */
-		irqnum = EXTI0_IRQn + line;
-	}
-#elif CONFIG_SOC_SERIES_STM32F4X
-	if (line >= 5 && line <= 9) {
-		irqnum = EXTI9_5_IRQn;
-	} else if (line >= 10 && line <= 15) {
-		irqnum = EXTI15_10_IRQn;
-	} else if (line >= 0 && line <= 4) {
-		/* pins 0..4 are mapped to EXTI0.. EXTI4 */
-		irqnum = EXTI0_IRQn + line;
-	} else {
-		switch (line) {
-		case 16:
-			irqnum = PVD_IRQn;
-			break;
-		case 17:
-			irqnum = RTC_Alarm_IRQn;
-			break;
-		case 18:
-			irqnum = OTG_FS_WKUP_IRQn;
-			break;
-		case 21:
-			irqnum = TAMP_STAMP_IRQn;
-			break;
-		case 22:
-			irqnum = RTC_WKUP_IRQn;
-			break;
-		}
-	}
-#elif CONFIG_SOC_SERIES_STM32F7X
-	if (line >= 5 && line <= 9) {
-		irqnum = EXTI9_5_IRQn;
-	} else if (line >= 10 && line <= 15) {
-		irqnum = EXTI15_10_IRQn;
-	} else if (line >= 0 && line <= 4) {
+	} else if {
```

Comment:
Looks really weird!

Reply:
Yes, I missed some code in the pushed commit, fixed Txs